migration/migrator/migrations/course/20210610100802_course_materials_json_to_db.py

```diff
@@ -0,0 +1,121 @@
+"""Migration for a given Submitty course database."""
+
+import json
+from pathlib import Path
+from sqlalchemy import insert
+
+def up(config, database, semester, course):
+    """
+    Run up migration.
+
+    :param config: Object holding configuration details about Submitty
+    :type config: migrator.config.Config
+    :param database: Object for interacting with given database for environment
+    :type database: migrator.db.Database
+    :param semester: Semester of the course being migrated
+    :type semester: str
+    :param course: Code of course being migrated
+    :type course: str
+    """
+    # create tables here
+    database.execute(
+        """
+        CREATE TABLE IF NOT EXISTS course_materials (
+            path TEXT PRIMARY KEY,
```

Comment:
need to update the `course_db.sql` with this new name. Also, this should probably be `varchar(255)`, unless you're expecting very very long paths?

Reply:
I was considering varchar(255) but wasn't sure if any instructor would ever have a subdirectory in addition to the normal directory that exceeds 255 chars. It is a limit we could establish and force. I know a 255 char path is already very long but if anyone was to hit it a warning would need to be shown. Thoughts?

Reply:
Something to consider is that the maximum file path length on Windows is 255 characters. Given a zip of all course materials, if the path exceeds that value, then people on Windows would never be able to extract all files from that zip.

So we probably actually want to set the limit for max path _lower_ than 255 to help protect against that case, though I don't think we ever considered that in the original implementation.